great_expectations/data_context/store/ge_cloud_store_backend.py

```diff
@@ -227,7 +263,7 @@ def allowed_set_kwargs(self) -> Set[str]:
             self.ge_cloud_resource_type, set()
         )
 
-    def validate_set_kwargs(self, kwargs: dict) -> Optional[bool]:
+    def validate_set_kwargs(self, kwargs: dict) -> Union[bool, None]:
```

Comment:
Should we replace all instances here? Perhaps something to discuss with the team

Reply:
Yes, I think this along with https://github.com/great-expectations/great_expectations/pull/5738 should be part of the custom linter/style-checker.

I will update our confluence typing page explaining these styles and why we want to use them.
Open for discussion/debate of course.